newrelic-agent/src/main/java/com/newrelic/agent/errors/ErrorDataImpl.java

```diff
@@ -0,0 +1,131 @@
+        if (isTransactionDataThrowableValid() && transactionData.getThrowable().throwable != null) {
+            return transactionData.getThrowable().throwable.getClass().toString();
+        } else if (isTracedErrorInstanceOfThrowableError()) {
+            return ((ThrowableError) tracedError).getThrowable().getClass().toString();
+        }
+
+        return "";
+    }
+
+    @Override
+    public String getErrorMessage() {
+        if (isTransactionDataThrowableValid() && transactionData.getThrowable().throwable.getMessage() != null) {
+            return transactionData.getThrowable().throwable.getMessage();
+        } else if (isTracedErrorInstanceOfThrowableError() && ((ThrowableError) tracedError).getThrowable() != null) {
+            return ((ThrowableError) tracedError).getThrowable().getMessage();
+        }
+        return "";
+    }
+
+    @Override
+    public StackTraceElement[] getStackTraceElement() {
+        if (isTransactionDataThrowableValid() && transactionData.getThrowable().throwable != null) {
+            return transactionData.getThrowable().throwable.getStackTrace();
+
+        } else if (isTracedErrorInstanceOfThrowableError() && ((ThrowableError) tracedError).getThrowable() != null) {
+            return ((ThrowableError) tracedError).getThrowable().getStackTrace();
+
+        }
+        return new StackTraceElement[] {};
+    }
+
+    @Override
+    public Map<String, ?> getCustomAttributes() {
+        Map<String, ? super Object> combinedUserAttMap = new HashMap<>();
+        if (transactionData != null) {
+            if (transactionData.getUserAttributes() != null) {
+                combinedUserAttMap.putAll(transactionData.getUserAttributes());
+            }
+            if (transactionData.getErrorAttributes() != null) {
+                combinedUserAttMap.putAll(transactionData.getErrorAttributes());
+            }
+        }
+        if (tracedError != null && tracedError.getErrorAtts() != null) {
+            combinedUserAttMap.putAll(tracedError.getErrorAtts());
+        }
+
+        return combinedUserAttMap;
+    }
+
+    @Override
+    public String getTransactionName() {
+        return transactionData != null ? transactionData.getTransaction().getPriorityTransactionName().getName() : "";
+    }
+
+    @Override
+    public String getTransactionUiName() {
+        return getTransactionName();
+    }
+
+    @Override
+    public String getRequestUri() {
+        return transactionData != null ? transactionData.getAgentAttributes().get(AttributeNames.REQUEST_URI).toString() : "";
+
+    }
+
+    @Override
+    public String getHttpStatusCode() {
+        return transactionData != null ? transactionData.getAgentAttributes().get(AttributeNames.HTTP_STATUS_CODE).toString() : "";
+    }
+
+    @Override
+    public String getHttpMethod() {
+        return transactionData != null ? transactionData.getAgentAttributes().get(AttributeNames.REQUEST_METHOD_PARAMETER_NAME).toString() : "";
```

Comment:
Does `transactionData.getAgentAttributes().get(AttributeNames. REQUEST_METHOD_PARAMETER_NAME)` ever return null? If it does return null, maybe add a default empty string value?